main.py

```diff
@@ -88,23 +89,23 @@ def main():
 
                     logger.debug(f'Data for sell: {coin=} | {stored_price=} | {coin_tp=} | {coin_sl=} | {volume=} | {symbol=} ')
 
-                    logger.info(f'get_last_price existing coin: {coin}')
-                    last_price = get_last_price(symbol, pairing)
+                    logger.info(f'get_coin_info existing coin: {coin}')
+                    coin_info = get_coin_info(symbol, pairing)
 
-                    logger.info("Finished get_last_price")
-                    logger.info(f'{last_price=}')
+                    logger.info("Finished get_coin_info")
+                    logger.info(f'{coin_info.last=}')
                     logger.info(f'{stored_price + (stored_price*sl /100)=}')
 
                     # update stop loss and take profit values if threshold is reached
-                    if float(last_price) > stored_price + (
+                    if float(coin_info.last) > stored_price + (
```

Comment:
Here you call the last price with coin_info.last.  You can use coin_info.xxx to pull different coin info.  

See https://www.gate.io/docs/apiv4/en/#retrieve-ticker-information for the different options.